### PR TITLE
Fix default/empty values for component prefill configuration

### PR DESCRIPTION
**Pull request by sergei-maertens**
Partly fixes open-formulieren/open-forms#3922

### CODE REVIEW

File: src/components/formio/select.tsx
Comment (sergei-maertens) [inline src/components/formio/select.tsx:1]: the typescript casting in this file is horrible... I remember dealing with react-select and their generics as one of the first typescript things I did and we definitely need to revisit this, but I'm putting that out of scope for this bugfix right now.

File: src/components/formio/select.tsx [hunk 1]
```diff
@@ -202,10 +211,16 @@ function Select<
...
             const isSingle = !Array.isArray(newValue);
-            const normalized = isSingle ? [newValue] : newValue;
-            const rawValues = normalized.map(val => val?.[valueProperty] ?? null);
+
+            // for multiple, the value is reset to empty array (`[]`)
+            if (action.action === 'clear' && isSingle) {
+              newValue = emptyValue;
+            }
+
+            const normalized = (isSingle ? [newValue] : newValue) as MultiValue<Option>;
```
Comment (vaszig) [inline src/components/formio/select.tsx:222]: Why do we explicitly declare the result type here?

**Review by vaszig — APPROVED** [review]

File: src/components/formio/select.tsx (hunk 1 above)
Comment (sergei-maertens) [reply src/components/formio/select.tsx:222]: otherwise the typescript compiler complains about implicit `any` types :)

**Comment by sergei-maertens** [comment]
Backport: a74a2d354dc769bbe168bf35c2c0d2fe089892f5

**Pull request merged**